src/transformStateWithClones.js

```diff
@@ -7,7 +7,56 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const ACTION_TYPES = {
+    clear: 'clear',
+    addProperties: 'addProperties',
+    removeProperties: 'removeProperties',
+  };
+  const states = [];
+
+  for (let i = 0; i < actions.length; i++) {
+    const currentAction = actions[i];
+    const currentState = states[i - 1] ?? state;
+
+    switch (currentAction.type) {
```

Comment:
Need to use the destruction for action
```suggestion
    const { type, ......... } = action;

    switch (type) {
```